cache/lru/cache.go

```diff
@@ -55,15 +72,19 @@ func (c *Cache[K, V]) Get(key K) (V, bool) {
 func (c *Cache[K, _]) Evict(key K) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-
+	value, ok := c.elements.Get(key)
+	if !ok {
+		return
+	}
+	c.onEvict(key, value)
 	c.elements.Delete(key)
```

Comment:
Sorry if my previous comment was vague - i meant to refactor this blob of code into a non-exported `evict` so that the rest of the cache code could call `evict` when it needed to remove an element + have the `onEvict` callback called.